plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/KtlintPlugin.kt

```diff
@@ -135,19 +140,74 @@ open class KtlintPlugin : Plugin<Project> {
+                        setCheckTaskDependsOnKtlintCheckTask(project, checkTask)
+
+                        val kotlinSourceSet = sourceDirectoriesList.reduce { acc, fileCollection ->
+                            acc.plus(fileCollection)
+                        }
+                        val runArgs = kotlinSourceSet.files.map { "${it.path}/**/*.kt" }.toMutableSet()
+                        addAdditionalRunArgs(extension, runArgs)
+
+                        val ktlintSourceSetFormatTask = createFormatTask(
+                            project,
+                            konanBuildingConfig.name,
+                            ktLintConfig,
+                            kotlinSourceSet,
+                            runArgs
+                        )
+                        addKtlintFormatTaskToProjectMetaFormatTask(project, ktlintSourceSetFormatTask)
+                    }
+                }
+            }
+        }
+    }
+
+    private fun applyKtLintNative(
+        project: Project,
+        extension: KtlintExtension
+    ): (AppliedPlugin) -> Unit {
+        return { _ ->
+            project.afterEvaluate { _ ->
+                val ktLintConfig = createConfiguration(project, extension)
+
+                project.components.withType(KotlinNativeComponent::class.java) { component ->
+                    val sourceDirectoriesList = mutableListOf<FileCollection>()
+                    component.konanTargets.get().forEach {
+                        sourceDirectoriesList.add(component.sources.getAllSources(it))
+                    }
+
+                    if (sourceDirectoriesList.isNotEmpty()) {
+                        val checkTask = createCheckTask(
+                            project,
+                            extension,
+                            component.name,
+                            ktLintConfig,
+                            sourceDirectoriesList
+                        )
                         addKtlintCheckTaskToProjectMetaCheckTask(project, checkTask)
                         setCheckTaskDependsOnKtlintCheckTask(project, checkTask)
 
                         val kotlinSourceSet = sourceDirectoriesList.reduce { acc, fileCollection ->
-                            acc.add(fileCollection)
+                            acc.plus(fileCollection)
                         }
                         val runArgs = kotlinSourceSet.files.map { "${it.path}/**/*.kt" }.toMutableSet()
                         addAdditionalRunArgs(extension, runArgs)
 
-                        val ktlintSourceSetFormatTask = createFormatTask(project, it.name, ktLintConfig,
-                                kotlinSourceSet, runArgs)
+                        val ktlintSourceSetFormatTask = createFormatTask(
+                            project,
+                            component.name,
+                            ktLintConfig,
+                            kotlinSourceSet,
+                            runArgs
+                        )
```

Comment:
Anything you can do to de-duplicate this code a bit more?

Reply:
It seems like this is the first time you're doing the duplication. Feel free to tackle it now instead of a seperate PR, even if the diff is more ugly, it will help with future maintainability.